src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DocumentClassifierPassBase.cs

```diff
@@ -1,27 +1,28 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
 using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Collections.Immutable;
 using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
 using Microsoft.AspNetCore.Razor.Language.Intermediate;
 
 namespace Microsoft.AspNetCore.Razor.Language;
 
 public abstract class DocumentClassifierPassBase : IntermediateNodePassBase, IRazorDocumentClassifierPass
 {
+    private ImmutableArray<ICodeTargetExtension> _targetExtensions;
+
     protected abstract string DocumentKind { get; }
 
-    protected IReadOnlyList<ICodeTargetExtension> TargetExtensions { get; private set; }
+    protected ImmutableArray<ICodeTargetExtension> TargetExtensions
+        => _targetExtensions.NullToEmpty();
```

Comment:
>=> _targetExtensions.NullToEmpty(); [](http://example.com/codeflow?start=8&length=35)

nit: could just initialize to empty and then not worry about null

Reply:
I suppose. It's really just for sanity.